### ISSUE TITLE
Multiple metrics in evaluate, ForecastingGridSearchCV [ENH]

### ISSUE THREAD

**Issue by anneserious** [sktime/sktime#3508]
**Is your feature request related to a problem? Please describe.**
I would like to measure various metrics during cross-validation.

**Describe the solution you'd like**
`evaluate` and `ForecastingGridSearchCV` argument `scoring` could accept a list of metrics which will all be be computed during evaluation. For `ForecastingGridSearchCV` would need an additional argument to indicate which matrics is to be used for ranking purposes.
This functionality is analogous to sklearn `sklearn.model_selection.GridSearchCV`.

**Comment by fkiraly** [comment]
FYI @DBCerigo, @TNTran92

**Comment by topher-lo** [comment]
Will to be good to plan this alongside the parallelization of `evaluate`

**Comment by hazrulakmal** [comment]
Correct me if I'm wrong but I think `evaluate` currently supports multiple metrics to be computed. for example by running the following code (fetched from the docs)

```python
y = load_airline()
forecaster = NaiveForecaster(strategy="mean", sp=12)
cv = ExpandingWindowSplitter(initial_window=12, step_length=3,

from sktime.performance_metrics.forecasting import MeanSquaredError
results = evaluate(
    forecaster=forecaster,
    y=y,
    cv=cv,
    scoring=[MeanSquaredError(square_root=True), MeanAbsoluteError()],
)
```

It should be possible to extend the capabilities of ForecastingGridSearchCV to include more than one metric in the scoring argument. Mainly because,  under the hood, the class uses `evaluate`. If this feature is needed, I'm free to take this off.

**Comment by hazrulakmal** [comment]
I have a question about interface design regarding the specification of which metrics to fit. There are a few options in my mind right now. would be great to have a second opinion.

The first option is to *create an additional argument* that specifies which metrics to choose. This could be one of the following:
1. The metric object itself, like metric_to_fit = MeanAbsoluteError()
2. A string/integer specifying the index of the metric, like metric_to_fit = 0 to indicate that fitting should be based on the first metric in the list.
3. A string specifying the name of the metric, like metric_to_fit = "mae". This would require us to implement a utility function, `get_scorer`, that maps a string to its respective metric class.

Another option is to *use one of the existing arguments* to avoid cluttering the interface. One idea is to perhaps extend the scoring argument to receive a dictionary for when multiple metrics are needed. an example would be in the form of `{"metrics"= [mse, mae, mape], "metric_to_fit" = "mae"}`. Ofcourse this would need us to implement a utility function `get_scorer`

I think I would prefer the second option because we dont need to add any additional argument. Also, if we implement the `get_scorer` function, it would be very convenient for users. Instead of importing the metric class and specifying it to any class that requires a scoring argument, they can just write the name of the metrics as a string and it will be handled internally. Perhaps, we could also extend the functionality so that users can check available metrics in sktime.

**Comment by hazrulakmal** [comment]
(just thought of it) Alternatively, similar to option two, users can provide a list of metrics as an argument to the scoring parameter (no need for a dict actually). The model will always be fitted based on the metric at the first index of the list. Therefore, users should rearrange the list to ensure that the first metric is the one they want the model to be fitted based on.

**Comment by hazrulakmal** [comment]
but that would mean users will lose the choice of having the best forecaster (`best_forecaster_` attribute) remains fitted on the last fold in cv, i.e refit=False. because in other to follow `GridSearchCV`, letting the refit argument accept a metric string, we have then to assume the boolean to be True.

**Comment by yarnabrina** [comment]
I'm not suggesting `refit` to be string-only. It's type can be `typing.Union[bool, str]`.

You can add a validation like this:

- If `scoring` is of length `1`, `refit` will be expected to be `bool`.
- If `scoring` is of length `>1`, `refit` has to be a string to decide what to rank by.

I think `scikit-learn` follows the same format, if you take a look at the documentation link I shared above and/or the notebook examples.

With very similar design and argument names, my preference will be to not vary much unless what we have to change to take time series into account.
